## **1. State the Hypotheses:**
Null Hypothesis (H₀): There is no association between the type of smart home device and customer satisfaction level.

Alternative Hypothesis (H₁): There is an association between the type of smart home device and customer satisfaction level.

## **2. Compute the Chi-Square Statistic:**

In [21]:
from scipy import stats
import pandas as pd
import numpy as np

observed = {
        'Satisfaction': ['Very Satisfied', 'Satisfied', 'Neutral', 'Unsatisfied', 'Very Unsatisfied'],
        'Smart Thermostat': [50, 80, 60, 30, 20],
        'Smart Light': [70, 100, 90, 50, 50]}

df = pd.DataFrame(observed)

df.set_index('Satisfaction', inplace=True)

print("Given Data: \n")
print(df)

row_total = df.sum(axis=1)
col_total = df.sum(axis=0)
grand_total = df.values.sum()

print("\nRow Total :")
print(row_total)
print("\nColumn Total :")
print(col_total)
print(f"\nGrand Total : {observed}")

expected = np.outer(row_total, col_total) / grand_total

expected_df = pd.DataFrame(expected,
                           columns = ['Smart Thermostat', 'Smart Light'],
                           index=df.index)

print("\nExpected Frequencies :")
print(expected_df)

chi_square_statistic, p_value, dof, expected = stats.chi2_contingency(df.values)
print(f"\nChi-Square Statistic: {chi_square_statistic}")
print(f"Degrees of Freedom: {dof}")


Given Data: 

                  Smart Thermostat  Smart Light
Satisfaction                                   
Very Satisfied                  50           70
Satisfied                       80          100
Neutral                         60           90
Unsatisfied                     30           50
Very Unsatisfied                20           50

Row Total :
Satisfaction
Very Satisfied      120
Satisfied           180
Neutral             150
Unsatisfied          80
Very Unsatisfied     70
dtype: int64

Column Total :
Smart Thermostat    240
Smart Light         360
dtype: int64

Grand Total : {'Satisfaction': ['Very Satisfied', 'Satisfied', 'Neutral', 'Unsatisfied', 'Very Unsatisfied'], 'Smart Thermostat': [50, 80, 60, 30, 20], 'Smart Light': [70, 100, 90, 50, 50]}

Expected Frequencies :
                  Smart Thermostat  Smart Light
Satisfaction                                   
Very Satisfied                48.0         72.0
Satisfied                     72.0        108.0
Neutral

## **Step 3: Determine the Critical Value**

In [22]:
alpha = 0.05
critical_value = stats.chi2.ppf(1 - alpha, dof)

print(f"Critical Value at alpha={alpha}: {critical_value}")

Critical Value at alpha=0.05: 9.487729036781154


## **Step 4: Make a Decision**

In [23]:
reject_null

if chi_square_statistic > critical_value:
    print("Reject the null hypothesis. There is a significant association between device type and customer satisfaction.")
else:
    print("Fail to reject the null hypothesis. There is no significant association between device type and customer satisfaction.")

Fail to reject the null hypothesis. There is no significant association between device type and customer satisfaction.


## **Report**

Assignment Tasks:
1. State the Hypotheses:
   - Null Hypothesis (H0): There is no significant association between the type of device purchased and the customer's satisfaction level.
   - Alternative Hypothesis (H1): There is a significant association between the type of device purchased and the customer's satisfaction level.

2. Compute the Chi-Square Statistic:
   - Chi-Square Statistic: **5.6382**
   - p-value: **0.2278**

3. Determine the Critical Value:
   - Degrees of Freedom: **4**
   - Critical Value (at alpha = 0.05) : **9.4877**

4. Make a Decision:
   - Conclusion: Fail to reject the null hypothesis. There is no significant association between device type and customer satisfaction.

Based on the calculated Chi-Square statistic and the critical value, we have conclusion :

**The type of device, whether it's a Smart Thermostat or Smart Light, does not significantly affect customer satisfaction.**